Core/Src/hexapod_spi_driver.c

```diff
@@ -0,0 +1,126 @@
+#include "hexapod_spi_driver.h"
+#include "spi.h"
+#include <stdbool.h>
+#include <malloc.h>
+#include <memory.h>
+#include "servo_control.h"
+
+
+static TIM_HandleTypeDef* servo_timers[] = {
+        SERVO_11_TIMER,
+        SERVO_12_TIMER,
+        SERVO_13_TIMER,
+        SERVO_21_TIMER,
+        SERVO_22_TIMER,
+        SERVO_23_TIMER,
+        SERVO_31_TIMER,
+        SERVO_32_TIMER,
+        SERVO_33_TIMER,
+        SERVO_41_TIMER,
+        SERVO_42_TIMER,
+        SERVO_43_TIMER,
+        SERVO_51_TIMER,
+        SERVO_52_TIMER,
+        SERVO_53_TIMER,
+        SERVO_61_TIMER,
+        SERVO_62_TIMER,
+        SERVO_63_TIMER
+};
+
+static uint8_t servo_channels[] = {
+        SERVO_11_CHANNEL,
+        SERVO_12_CHANNEL,
+        SERVO_13_CHANNEL,
+        SERVO_21_CHANNEL,
+        SERVO_22_CHANNEL,
+        SERVO_23_CHANNEL,
+        SERVO_31_CHANNEL,
+        SERVO_32_CHANNEL,
+        SERVO_33_CHANNEL,
+        SERVO_41_CHANNEL,
+        SERVO_42_CHANNEL,
+        SERVO_43_CHANNEL,
+        SERVO_51_CHANNEL,
+        SERVO_52_CHANNEL,
+        SERVO_53_CHANNEL,
+        SERVO_61_CHANNEL,
+        SERVO_62_CHANNEL,
+        SERVO_63_CHANNEL
+};
+
+static bool isFrameType(uint8_t frame_length, uint8_t suspected_frame_length){
+    if(frame_length == suspected_frame_length){
+        return true;
+    }
+    return false;
+}
```

Comment:
Nie łatwiej: `return frame_length == suspected_frame_length` ?

Reply:
Racja, możesz wrzucić tą zmianę albo ja to wrzucę.